bindings/pydeck/pydeck/io/html.py

```diff
@@ -87,14 +88,30 @@ def display_html(filename):
 
 
 def iframe_with_srcdoc(html_str, width="100%", height=500):
-    width = '"{}"'.format(width) if type(width) == str else width
-    iframe = """<iframe src="about:blank" frameborder="0" srcdoc="{}" width={} height={}></iframe>""".format(
-        html.escape(html_str), width, height
+    if isinstance(width, str):
+        width = '"{}"'.format(width)
+
+    iframe = """
+        <iframe
+            width={width}
+            height={height}
+            frameborder="0"
+            srcdoc="{srcdoc}"
+        ></iframe>
+    """
+
+    iframe = textwrap.dedent(iframe).strip()
+    iframe = iframe.format(
+        width=width,
+        height=height,
+        srcdoc=html.escape(html_str),
     )
+
     from IPython.display import HTML  # noqa
 
     with warnings.catch_warnings():
-        warnings.filterwarnings("ignore", message="Consider using IPython.display.iframe instead")
+        msg = "Consider using IPython.display.iframe instead"
```

Comment:
Should be capitalized ([doc](https://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html#IPython.display.IFrame))
```suggestion
        msg = "Consider using IPython.display.IFrame instead"
```

Reply:
Though on second read, this is catching the warning not propagating it, so this should also be fixed wherever the warning is created.

Reply:
This is created within the IPython module itself. https://github.com/ipython/ipython/blob/6760a46eb17a32ce1c960e1de0534594bcf03267/IPython/core/display.py#L413

It's supposed to be suppressed–we want the `srcdoc`attribute, which isn't made available by the IFrame object.